library/include/borealis/core/input.hpp

```diff
@@ -66,11 +69,34 @@ enum ControllerAxis
 };
 
 // Represents the state of the controller (a gamepad or a keyboard) in the current frame
-typedef struct ControllerState
+struct ControllerState
 {
     bool buttons[_BUTTON_MAX]; // true: pressed
     float axes[_AXES_MAX]; // from 0.0f to 1.0f
-} ControllerState;
+};
+
+// Represents a touch phase in the current frame
+enum class TouchPhase
+{
+    START,
+    STAY,
+    END,
+    NONE,
+};
+
+// Contains raw touch data
+struct RawTouch
```

Comment:
I'm not sure about the name of the two structs here, "touch" does not represent anything

RawTouch could be named RawTouchState since it represents the state of touch in the lower level (platform driver).

 And so Touch could be named TouchState